### DISCUSSION TITLE
`useHead` callback

### DISCUSSION THREAD

**Discussion by jolo-dev** [nuxt/nuxt#16424] [Questions]
Afaik, Nuxt is using `https://vue-meta.nuxtjs.org/` with composable `useHead`.
But I guess the [`callback`](https://vue-meta.nuxtjs.org/api/#callback) is not working.

Here is my codesnippet

```ts
useHead({
    script: [
      {
        src: 'https://accounts.google.com/gsi/client',
        defer: true,
        async: true,
        callback: () => {
          console.log('Loading finished')
        }
      }
    ]
  })
```

I expect to have a `console.log('Loading finished')` as I want to trigger an action once the script has been loaded.
Perhaps, I do something wrong or anyone can confirm that?

**Comment by zihaoyy** [comment]
> Afaik, Nuxt is using `https://vue-meta.nuxtjs.org/` with composable `useHead`. But I guess the [`callback`](https://vue-meta.nuxtjs.org/api/#callback) is not working.
> 
> Here is my codesnippet
> 
> ```ts
> useHead({
>     script: [
>       {
>         src: 'https://accounts.google.com/gsi/client',
>         defer: true,
>         async: true,
>         callback: () => {
>           console.log('Loading finished')
>         }
>       }
>     ]
>   })
> ```
> 
> I expect to have a `console.log('Loading finished')` as I want to trigger an action once the script has been loaded. Perhaps, I do something wrong or anyone can confirm that?

Currently using `useHead` does not seem to have this function, a better suggestion is to use `onMounted` to operate after the page is rendered

**Reply by JackEdwardLyons** [reply]
Is it better to simply use the `head()` method instead? Or will that be deprecated in Nuxt 3?

**Comment by TheAlexLichter** [comment]
Created an issue in https://github.com/vueuse/head/issues/156 ☺️
